### DISCUSSION TITLE
arguments autocomplete with cake.tool

### DISCUSSION THREAD

**Discussion by MihaMarkic** [cake-build/cake#2973] [Ideas]
Is there a way to have arguments autocomplete with cake.tool?

**Comment by gep13** [comment]
Just to make sure we are talking about the same thing... are you talking about the direct arguments into Cake itself, or the arguments that you define within your Cake script?

**Reply by MihaMarkic** [reply]
both actually. I'd like to have autocomplete when I type `dotnet cake`

**Comment by pascalberger** [comment]
Autocomplete would be a nice feature to have. We probably need to differentiate between the CLI switches and the name of the tasks defined in the build script. 

For Cake .NET tool it should be possible to implement autocomplete for the CLI switches. `--target` is here not a CLI switch, but something which is defined by the build script, and can have any name. Support the argument for passing task names and the names itself is more complicated. Also performance is an issue since we don't want to recompile on every key stroke. This might be doable by caching result somewhere.

For Cake Frosting `--target` is a CLI switch and it should be possible to support it as also support for task names.

**Reply by pascalberger** [reply]
I created https://github.com/cake-build/cake/issues/2974 for tracking tab completion.
